cli/tests/test_observability_v8_upgrade_migration.py

```
@@ -64,6 +73,506 @@ def test_missing_config_is_an_unconfigured_installation_no_op(self) -> None:
+                    candidate_directory=self.root.name,
+                )
+        self.assertEqual(raised.exception.code, "source_read_failed")
+        convert.assert_not_called()
+
+    @unittest.skipUnless(hasattr(os, "mkfifo"), "FIFO creation is POSIX-only")
+    def test_preflight_rejects_nonregular_config_without_blocking(self) -> None:
+        os.unlink(self.config_path)
+        os.mkfifo(self.config_path)
+        convert = Mock()
+        with patch("defenseclaw.migrations.convert_v7_observability_to_v8", convert):
+            with self.assertRaises(ObservabilityV8UpgradeMigrationError) as raised:
```

Ruff (0.15.21)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)